doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -433,6 +433,18 @@ implementation, the tactic will always choose the first possible
 solution to the set of constraints generated by a rewrite and will not
 try to find *all* the possible solutions to warn the user about them.
 
+Rewriting with ``Type`` valued relations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
Is this the best placement for the added information?  It's hard for me to gauge how important this and how often users will use it.  We like to put more important and more commonly used items first (subject to compromise to describe features in a coherent order).

Reply:
I agree that it might not be the best place, but I think this whole documentation is in need of a rewrite. It has been adapted from a long-gone legacy system, and I think we need to sit down and remove a lot of the superfluous information present. For the scope of this PR, I needed to mention somewhere that generalized rewriting does work along Type valued relations, whereas before no mention of this was present. Still users were experimenting hence all the bugs that will be closed by this PR. I'll see if I can find a better place still.

Reply:
I think the location is still best since the parts before describe using `rewrite` in differing contexts. This is an example of a different context (when the relation is Type valued). https://coq.gitlab.io/-/coq/-/jobs/1940975675/artifacts/_install_ci/share/doc/coq/sphinx/html/addendum/generalized-rewriting.html#rewriting-with-type-valued-relations

Reply:
> but I think this whole documentation is in need of a rewrite.

Probably.  It would be nice to understand how "generalized rewriting" differs from "rewriting" (if relevant to users).  Also defining "morphism" for the benefit of those who are not among the cognoscenti.

> I think we need to sit down and remove a lot of the superfluous information present

And add some material, too.  Let me know if you want to work together on improving the chapter (in a separate PR).  I can work on the mechanics of writing, but I would need participation by a subject-matter expert to make any progress.